src/@providers/docker/services/database.ts

```diff
@@ -239,15 +248,24 @@ export class DockerDatabaseService extends CrudResourceService<'database', Docke
           existingDeployment.labels?.['io.architect.arcctl.databaseVersion'],
       },
       exposed_ports: [{
-        port: 5432,
         target_port: 5432,
       }],
     });
 
+    const inspectionRes = await this.deploymentService.inspect(deployment.id);
+    if (!inspectionRes) {
+      throw new Error(`Database failed to start`);
+    }
+
+    const hostPort = inspectionRes.NetworkSettings.Ports?.['5432/tcp']?.[0].HostPort;
+    if (!hostPort) {
+      throw new Error('Failed to allocate a port to listen on');
+    }
+
     return {
       id: normalizedName,
       host: 'host.docker.internal',
-      port: 5432,
+      port: Number(hostPort),
       username: 'architect',
       password: 'architect',
       protocol: 'postgresql',
```

Comment:
Why does redis have the postgres protocol?

Reply:
Fixed. Forgot to update the `update` method.